File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        full_prod = 1
        numZero = 0
        out = []
        for n in nums:
            if n == 0:
                numZero += 1
            else:
                full_prod *= n
        if numZero > 1:
            out = [0]*len(nums)
        elif numZero == 1:
            for n in nums:
                if n == 0:
                    out.append(full_prod)
                else:
                    out.append(0)
        else:
            for n in nums:
                    out.append(int(full_prod / n))
        return out
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::unordered_set<int> st(nums.begin(), nums.end());
        int longest = 0;
        for (auto n : st) {
            if (!st.contains(n-1)) {
                // Start of a sequence
                int next = ++n;
                int length = 1;
                while(st.contains(next)){ 
                    ++length;
                    ++next;
                }
                longest = max(longest, length);
            }
        }
        return longest;
    }
};
